enterprise_gateway/services/kernels/remotemanager.py

```diff
@@ -286,8 +346,20 @@ def remove_kernel(self, kernel_id):
         super().remove_kernel(kernel_id)
         self.parent.kernel_session_manager.delete_session(kernel_id)
 
+    def list_kernels(self, tenant_id: Optional[str] = UNIVERSAL_TENANT_ID):
+        """Returns a list of kernel_id's of kernels running."""
+        kernels = []
+        kernel_ids = self.tenant_kernels.get_kernel_ids(tenant_id)
```

Comment:
correct me, by doing this new way of `listing kernels`, we are introducing 2 different source of truth for running kernels list: 
1. Comes from the `MultiKernelManager::_kernels` which has been the single source of truth until now.
2. The new class ` TenantToKernels` which maintains the mapping with `tenant_id` -> `kernel_id` and vice versa. 

The challenge will be to always keep these 2 in sync.

Instead, we can iterate over each kernels and check if it matches the given `tenant_id`.

Reply:
> we can iterate over each kernels and check if it matches the given tenant_id.

Not sure what you mean here.  We do not store the tenant_id in the `KernelManager`. Only the `RemoteMappingKernelManager` knows about tenant_id.

Reply:
After a sidebar discussion, I'll push tenant_id onto `RemoteKernelManager` and we'll continue to leverage `self._kernels` as the single source of truth and discard the management of the mapping lists.